packages/eslint-config/.eslintrc.js

```diff
@@ -0,0 +1,66 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/eslint-config
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+/**
+ * Default ESLint configuration for LoopBack
+ *
+ * See https://eslint.org/docs/user-guide/configuring
+ */
+module.exports = {
+  root: true,
+  parser: '@typescript-eslint/parser',
+  plugins: ['eslint-plugin', '@typescript-eslint'],
+  env: {
+    es6: true,
+    node: true,
+  },
+  extends: ['eslint:recommended', 'plugin:@typescript-eslint/recommended'],
+  rules: {
+    'comma-dangle': ['error', 'always-multiline'],
+    'no-mixed-operators': 'error',
+    'no-console': 'off',
+    'no-undef': 'off',
+    'no-inner-declarations': 'off',
+    // TypeScript allows method overloading
+    'no-dupe-class-members': 'off',
+    'no-useless-escape': 'off',
+    // TypeScript allows the same name for namespace and function
+    'no-redeclare': 'off',
+    '@typescript-eslint/indent': 'off',
+    '@typescript-eslint/no-explicit-any': 'error',
+    '@typescript-eslint/no-non-null-assertion': 'off',
+    '@typescript-eslint/explicit-function-return-type': 'off',
+    '@typescript-eslint/explicit-member-accessibility': 'off',
+    '@typescript-eslint/no-var-requires': 'off',
+    '@typescript-eslint/no-use-before-define': 'off',
+    '@typescript-eslint/no-object-literal-type-assertion': 'off',
+    '@typescript-eslint/no-parameter-properties': 'off',
+    '@typescript-eslint/no-angle-bracket-type-assertion': 'off',
+    '@typescript-eslint/prefer-interface': 'off',
+    '@typescript-eslint/no-namespace': 'off',
```

Comment:
I see a bunch of rules that we are disabling against the recommendations. What is your reasoning for disabling them? Also why is `ban-types` treated differently from these rules? Is it because fixing `ban-types` violations is easier than fixing violations of these other rules? That seems to be a poor excuse to me.

IMO, we should add `ban-type` to the list of disabled rules to keep this PR smaller and then incrementally try to enable most of the recommended rules, one by one. Starting with `ban-types` if you like.